Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -176,20 +175,26 @@ internal RetryHandler(HttpMessageHandler innerHandler) : base(innerHandler)
             {
                 HttpResponseMessage response = null;
                 int backOffInSec = 1;
+                int totalRetryTime = 0;
+                int maxDefaultBackoff = 16;
 
                 ServicePoint p = ServicePointManager.FindServicePoint(requestMessage.RequestUri);
                 p.Expect100Continue = false; // Saves about 100 ms per request
                 p.UseNagleAlgorithm = false; // Saves about 200 ms per request
                 p.ConnectionLimit = 20;      // Default value is 2, we need more connections for performing multiple parallel queries
 
-                TimeSpan httpTimeout = (TimeSpan)requestMessage.Properties["TIMEOUT_PER_HTTP_REQUEST"];
+                TimeSpan httpTimeout = (TimeSpan)requestMessage.Properties[SFRestRequest.HTTP_REQUEST_TIMEOUT_KEY];
+                logger.Debug("Http request timeout : " + httpTimeout);
+                TimeSpan restTimeout = (TimeSpan)requestMessage.Properties[SFRestRequest.REST_REQUEST_TIMEOUT_KEY];
+                logger.Debug("Rest request timeout : " + restTimeout);

```

Comment:
Use string format?

Reply:
Concatenation is supposed to be faster for .Net.
But I think it raises a point about the time we'll spend logging.
I added these log lines in debug, just in case with need to troubleshoot more issues, to make sure that the correct timeouts are propagated to the httpclient.

I did a micro benchmark and I will update the code shortly with the optimized version which will check if IsDebugEnabled.

With log level Debug
|              Method |        Mean |     Error |     StdDev |      
|-------------------- |------------:|----------:|-----------:|
|             WithLog |   856.41 ns | 28.928 ns |  84.840 ns |
|      CheckBeforeLog |    15.32 ns |  0.431 ns |   1.250 ns | 
| WithStringFormatLog | 1,440.98 ns | 88.273 ns | 258.889 ns | 

With log level in Error
|              Method |        Mean |     Error |    StdDev |
|-------------------- |------------:|----------:|----------:|
|             WithLog |   791.04 ns | 15.815 ns | 18.212 ns |
|      CheckBeforeLog |    12.00 ns |  0.244 ns |  0.342 ns |
| WithStringFormatLog | 1,166.14 ns | 22.869 ns | 32.798 ns |